### DISCUSSION TITLE
Files on demand - made a better title...

### DISCUSSION THREAD

**Discussion by MuhammadMusaabUlHaq** [abraunegg/onedrive#1565] [Q&A] [answered]
So, this is a simple question.
If I start onedrive, will it auto download all the files or I can download it according to my needs.
I dont have a linux to test it right now and I am just making a list of the things I need to have on linux cuz windows take 3 gigs of ram on my potato and its annoying to run linux on vm to get a terminal. And I just want to run linux as well so yeah...

**Comment by abraunegg** [comment, marked as answer]
@IamNotANoob 
> If I start onedrive, will it auto download all the files or I can download it according to my needs

The client will download, by default, all of your files. The client can also be configured to just download what you want. Please read the documentation. 

If you use the 'files-on-demand' feature within Windows OneDrive Client, you are better off using https://github.com/jstaf/onedriver

**Comment by MuhammadMusaabUlHaq** [comment]
Thank you very much!

**Comment by MuhammadMusaabUlHaq** [comment]
Though can I configure this one drive before enabling it? so it only download files on demand?
Or is it impractical?

**Comment by abraunegg** [comment]
@IamNotANoob 
> Though can I configure this one drive before enabling it? so it only download files on demand?

You can configure this client post installation and before first use, but if you are using Ubuntu (dont ..) you will encounter this issue - https://github.com/abraunegg/onedrive/issues/1274

Once configured, use `--dry-run` to *test* your configuration before using it.

If you are using files on-demand, this client does *not* support that - use https://github.com/jstaf/onedriver

**Reply by ezhan** [reply]
First of all, thanks for your work! 

About the on-demand feature: 
1. My onedrive is 75 Gb. My Linux partition is 80 Gb. So I have to use on-demand feature.
2. I have read the documentation about enabling only what I want. It is very complicated for a beginner. 
3. Is there plans for a GUI?

4. Onedriver project seems to be dead.

**Reply by abraunegg** [reply]
@ezhan 
> My onedrive is 75 Gb. My Linux partition is 80 Gb. So I have to use on-demand feature.

'on-demand' is not a feature of this client. It is currently a Feature Request. Please read: https://github.com/abraunegg/onedrive/issues/757 - this will be looked at, but not in the short term.

80GB is very small these days - it might be worth in you investing in larger local storage and/or partitioning your system to give more disk space.

> I have read the documentation about enabling only what I want. It is very complicated for a beginner.

It is not that complicated in reality - please read the documentation fully as this has examples.

> Is there plans for a GUI?

There already is a GUI - as per the documentation

> Onedriver project seems to be dead.

Unfortunately that is beyond my control. Potentially the author has decided that the Gnome 46+ integration provides better value (which it does not, and is horribly lacking capability), or the author no longer has any time to invest to support the gaps such are Shared Folders and Business / SharePoint capabilities.

**Reply by ezhan** [reply]
Linux is not my daily driver, I installed it to see how it works and does it suites me at all. That's why I have a 80 Gb partition for Linux. I even share Thunderbird profile with Windows, the profile is located on a NTFS partition.

Offtopic. I really hope KDE will become a main DE on Linux, not Gnome. Fedora 42 seems to offer KDE version alongside with Gnome. That is a big spet already.
